### PR TITLE
Add Animal data persistence feature.

### PR DESCRIPTION
**Pull request by euchangxian**
Implemented persistence of Animal Data.

### CODE REVIEW

**Review by Blizzeracz — COMMENTED** [review]
.

**Review by Blizzeracz — APPROVED** [review]
There are still remnants of Person class in the JavaDocs but all else is good. Storage class is well explained! Good job Pookie!

File: src/main/java/seedu/address/storage/JsonAdaptedAnimal.java [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+        this.petId = source.getPetIdForSerialization();
+        this.sex = source.getSexForSerialization();
+        this.species = source.getSpeciesForSerialization();
+        this.breed = source.getBreedForSerialization();
+        this.dateOfBirth = source.getDateOfBirthForSerialization();
+        this.admissionDate = source.getAdmissionDateForSerialization();
+    }
+
+    /**
+     * Converts this Jackson-friendly adapted person object into the model's {@code Person} object.
```
[the remark this answers is not in the document]
Comment (euchangxian) [reply src/main/java/seedu/address/storage/JsonAdaptedAnimal.java]: Resolved in https://github.com/AY2324S1-CS2103T-F08-3/tp/pull/87/commits/7da17c1e7e7b9ee87f09f48c6d6444da13ae0a79.

File: src/main/java/seedu/address/logic/AnimalLogicManager.java [hunk 2]
```diff
@@ -28,13 +30,13 @@ public class AnimalLogicManager implements AnimalLogic {
     private final Logger logger = LogsCenter.getLogger(AnimalLogicManager.class);
 
     private final AnimalModel model;
-    private final Storage storage;
+    private final AnimalStorage storage;
     private final AnimalCatalogParser animalCatalogParser;
 
     /**
-     * Constructs a {@code LogicManager} with the given {@code Model} and {@code Storage}.
+     * Constructs a {@code LogicManager} with the given {@code Model} and {@code AnimalStorage}.
```
[the remark this answers is not in the document]
Comment (euchangxian) [reply src/main/java/seedu/address/logic/AnimalLogicManager.java]: Resolved in https://github.com/AY2324S1-CS2103T-F08-3/tp/pull/87/commits/7da17c1e7e7b9ee87f09f48c6d6444da13ae0a79

File: src/main/java/seedu/address/MainApp.java [hunk 3]
```diff
@@ -71,24 +72,23 @@ public void init() throws Exception {
      * The data from the sample address book will be used instead if {@code storage}'s address book is not found,
      * or an empty address book will be used instead if errors occur when reading {@code storage}'s address book.
```
[the remark this answers is not in the document]
Comment (euchangxian) [reply src/main/java/seedu/address/MainApp.java]: Resolved in https://github.com/AY2324S1-CS2103T-F08-3/tp/pull/87/commits/7da17c1e7e7b9ee87f09f48c6d6444da13ae0a79

File: src/test/java/seedu/address/testutil/TypicalAnimals.java [hunk 4]
```diff
@@ -12,13 +12,46 @@
...
+        .withName("Muffin")
+        .withPetId("3333")
+        .withSex("MALE")
+        .withSpecies("Dog")
+        .withBreed("Jack Russell Terrier")
+        .withDateOfBirth("2020-10-10")
+        .withAdmissionDate("2020-12-12")
+        .build();
+
+    public static final Animal POOKIE = new AnimalBuilder()
```
[the remark this answers is not in the document]
Comment (euchangxian) [reply src/test/java/seedu/address/testutil/TypicalAnimals.java]: resolved in https://github.com/AY2324S1-CS2103T-F08-3/tp/pull/87/commits/28a8c6e9ae6d495880f64e760e5ec2d8a10bacdc

**Pull request merged**